user-service/controller/user-controller.js

```diff
@@ -19,3 +29,73 @@ export async function createUser(req, res) {
         return res.status(500).json({ message: 'Database failure when creating new user!' });
     }
 }
+export async function loginUser(req, res) {
+    // Check for existing username in db
+    const user = await getUser(req.body.username);
+
+    // if user does not exist
+    if (!user) {
+        return res.status(401).json({ message: 'Authentication failed. Invalid username.' });
+    }
+    const hashedPassword = user.password;
+    const isCorrectPassword = await verifyPassword(req.body.password, hashedPassword);
+
+    // if password mismatch
+    if (!isCorrectPassword) {
+        return res.status(401).json({ message: 'Authentication failed. Invalid password.' });
+    }
+
+    const token = await generateAccessToken(user);
+    const refreshToken = await generateRefreshAccessToken(user);
+    // Store new refresh token in db
+    refreshTokens.push(refreshToken);
+    return res.json({

```

Comment:
Perhaps can add a simple success message in the json body too, I was a little confused when I only saw the tokens

Reply:
Additionally, not sure what's the best practice, but I think it'll be clearer if we add `.status(200)` for all the OK responses too? to standardise with the other return results

Reply:
Agreed, updated to reflect these two suggestions